google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/appengine/sdk/ManagedCloudSdkServiceTest.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.sdk;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.tools.intellij.appengine.sdk.CloudSdkService.SdkStatus;
+import com.google.cloud.tools.intellij.testing.CloudToolsRule;
+import com.google.cloud.tools.intellij.testing.log.TestInMemoryLogger;
+import com.google.cloud.tools.intellij.util.ThreadUtil;
+import com.google.cloud.tools.managedcloudsdk.ManagedCloudSdk;
+import com.google.cloud.tools.managedcloudsdk.UnsupportedOsException;
+import com.google.cloud.tools.managedcloudsdk.command.CommandExecutionException;
+import com.google.cloud.tools.managedcloudsdk.components.SdkComponent;
+import com.google.cloud.tools.managedcloudsdk.components.SdkComponentInstaller;
+import com.google.cloud.tools.managedcloudsdk.install.SdkInstaller;
+import com.google.common.util.concurrent.MoreExecutors;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.concurrent.ExecutorService;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.Spy;
+
+/** Tests for {@link ManagedCloudSdkService} */
+public class ManagedCloudSdkServiceTest {
+
+  @Rule public CloudToolsRule cloudToolsRule = new CloudToolsRule(this);
+
+  private final Path MOCK_SDK_PATH = Paths.get("/tools/gcloud");
+
+  @Spy private ManagedCloudSdkService sdkService;
+
+  @Mock private ManagedCloudSdk mockManagedCloudSdk;
+
+  @Mock private CloudSdkService.SdkStatusUpdateListener mockStatusUpdateListener;
+
+  @Before
+  public void setUp() throws UnsupportedOsException {
+    doReturn(mockManagedCloudSdk).when(sdkService).createManagedSdk();
+    sdkService.setLogger(new TestInMemoryLogger());
```

Comment:
Is this custom logger still needed?

Reply:
Looks like we need this to avoid using standard IJ logging system - when it logs exceptions it prevents some tests to finish successfully. I plan to remove that when I fix and update the PR to use test logging factory for all `CloudToolsRule` tests.

Reply:
thats what I suspected. Its those `log#error` calls which throw exceptions on the non-test logger. Sounds good. Maybe a `todo` here to remove this?

Reply:
good idea, I'll add `todo`